packages/@aws-cdk/aws-servicecatalogappregistry/lib/private/validation.ts

```diff
@@ -0,0 +1,32 @@
+import * as cdk from '@aws-cdk/core';
+/**
+ * Class to validate that inputs match requirements.
+ */
+export class InputValidator {
+  /**
+   * Validates length is between allowed min and max lengths.
+   */
+  public static validateLength(resourceName: string, inputName: string, minLength: number, maxLength: number, inputString?: string): void {
+    if (!cdk.Token.isUnresolved(inputString) && inputString !== undefined && (inputString.length < minLength || inputString.length > maxLength)) {
+      throw new Error(`Invalid ${inputName} for resource ${resourceName}, must have length between ${minLength} and ${maxLength}, got: '${this.truncateString(inputString, 100)}'`);
+    }
+  }
+
+  /**
+   * Validates a regex.
+   */
+  public static validateRegex(resourceName: string, inputName: string, regex: RegExp, inputString?: string): void {
+    if (!cdk.Token.isUnresolved(inputString) && inputString !== undefined && !regex.test(inputString)) {
+      throw new Error(`Invalid ${inputName} for resource ${resourceName}, must match regex pattern ${regex}, got: '${inputString}'`);
```

Comment:
Not truncating this one?

Reply:
Didn't originally do it because it validates length first, but we truncate the string to 100 chars while the name can be longer, so probably would be better parity to do that hear as well.  Will add